src/libopensc/log.c

```diff
@@ -84,10 +84,21 @@ void sc_do_log_openssl(sc_context_t *ctx, int level, const char *file, int line,
 	}
 	ERR_print_errors(bio);
 
-	if ((length = BIO_pending(bio)) <= 0 ||
-			(buffer = malloc(length)) == NULL ||
-			BIO_read(bio, buffer, length) <= 0) {
-		sc_do_log(ctx, level, file, line, func, "Cannot log OpenSSL error");
+	length = BIO_pending(bio);
+	if (length <= 0) {
+		/* no error? */
+		goto end;
+	}
+	/* trailing null byte */
+	buffer = malloc(length + 1);
+	if (buffer == NULL) {
+		sc_do_log(ctx, level, file, line, func, "No memory!");
+		goto end;
+	}
+	rc = BIO_read(bio, buffer, length);
+	buffer[length] = '\n';
```

Comment:
This is not NUL terminated, this is a newline!

Reply:
:facepalm: will fix!